src/ebay-listbox/listbox.tsx

```diff
@@ -0,0 +1,217 @@
+
+            scroll(el)
+
+            if (wasClickedRef.current) {
+                wasClickedRef.current = false
+            }
+
+            handleChange(event, nextSelectedIndex, wasClicked)
+        }
+
+        if (options.length && !disabled) {
+            const container = containerRef.current
+            const optionsContainer = containerRef.current
+
+            activeDescendantRef.current = createLinear(
+                container,
+                optionsContainer,
+                optionsContainer,
+                '.listbox__option[role=option]',
+                {
+                    activeDescendantClassName: 'listbox__option--active',
+                    autoInit: selectedIndex,
+                    autoReset: null,
+                    autoScroll: listSelection !== 'auto'
+                }
+            )
+
+            typeaheadRef.current = typeahead()
+        }
+
+
+        if (listSelection === 'auto') {
+            containerRef.current.addEventListener('activeDescendantChange', handleListboxChange)
+        }
+
+        return () => {
+            if (activeDescendantRef.current) {
+                activeDescendantRef.current.reset()
+                activeDescendantRef.current.destroy()
+                activeDescendantRef.current = undefined
+            }
+
+            if (typeaheadRef.current) {
+                typeaheadRef.current.destroy()
+                typeaheadRef.current = undefined
+            }
+
+
+            if (containerRef.current) {
+                containerRef.current.removeEventListener('activeDescendantChange', handleListboxChange)
+            }
+        }
+    }, [selectedIndex, disabled, listSelection])
+
+    return (
+        <>
+            <div
+                {...rest}
+                tabIndex={tabIndex}
+                ref={containerRef}
+                role="listbox"
+                onKeyDown={listSelection !== 'auto' ? handleKeyDown : undefined}
+                className={classNames('listbox__options', className)}
+                style={{ maxHeight }}>
+                {options.map((option, index) =>
+                    cloneElement(option, {
+                        key: option.props.value || index,
+                        ...option.props,
+                        onMouseDown: listSelection === 'auto' ? handleMouseDown : undefined,
+                        onClick: (event: MouseEvent) => {
+                            if (listSelection !== 'auto') {
+                                handleChange(event, index, true)
+                            }
+                        },
+                        selected: index === selectedIndex
+                    })
+                )}
+            </div>
+
+            <select
+                hidden
+                className={classNames('listbox__native', selectClassName)}
+                name={name}
+                disabled={disabled}
+                value={options[selectedIndex]?.props.value}
```

Comment:
I believe we use `selected` in `<option>` here. Part of the reason is for form submission. When using value it doesn't seem to be working properly
Here's a test I made: 
https://jsfiddle.net/pfoe3zhn/

Reply:
Make sense.  Let me update

Reply:
@agliga this is true in HTML but not React. React has special handling for `<select value>`. See docs [here](https://react.dev/reference/react-dom/components/select#controlling-a-select-box-with-a-state-variable)

Reply:
In fact, based on the docs `selected` is not supported at all in React

Reply:
React throws a warning to use `value` or `defaultValue` instead of selected